docs/blocks/states.md

```diff
@@ -1,96 +1,149 @@
+Once you have your property constant, override `Block#createBlockStateDefinition(StateDefinition$Builder)` in your block class. In that method, call `StateDefinition.Builder#add(YOUR_PROPERTY);`. `StateDefinition.Builder#add` has a vararg parameter, so if you have multiple properties, you can add them all in one go.
+
+Every block will also have a default state. If nothing else is specified, the default state uses the default value of every property. You can change the default state by calling the `Block#registerDefaultState(BlockState)` method from your constructor.
+
+If you wish to change which `BlockState` is used when placing your block, override `Block#getStateForPlacement(BlockPlaceContext)`. This can be used to, for example, set the direction of your block depending on where the player is standing or looking when they place it.
+
+To further illustrate this, this is what the relevant bits of the `LeverBlock` class look like:
+
+```java
+public class LeverBlock extends Block {
+    // Note: It is possible to directly use the values in BlockStateProperties instead of referencing them here again. However, for the sake of simplicity and readability, it is recommended to add constants like this.
+    public static final DirectionProperty FACING = BlockStateProperties.FACING;
+    public static final BooleanProperty POWERED = BlockStateProperties.POWERED;
+
+    public LeverBlock(BlockBehaviour.Properties pProperties) {
+        super(pProperties);
+        registerDefaultState(stateDefinition.any() // stateDefinition.any() returns a random BlockState from an internal set, we don't care because we're setting all values ourselves anyway
+                .setValue(FACING, Direction.NORTH)
+                .setValue(POWERED, false)
+        );
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> pBuilder) {
+        pBuilder.add(FACING, POWERED); // this is where the properties are actually added to the state
+    }
+
+    @Override
+    @Nullable
+    public BlockState getStateForPlacement(BlockPlaceContext pContext) {
+        // code that determines which state will be used when placing down this block, depending on the BlockPlaceContext
+    }
+}
+```
+
+Using Blockstates
+-----------------
+
+To go from `Block` to `BlockState`, call `Block#defaultBlockState()`. The default blockstate can be changed through `Block#registerDefaultState`, as described above.
+
+You can get the value of a property by calling `BlockState#getValue(Property<?>)`, passing it the property you want to get the value of. Reusing our lever example, this would look something like this:
+
+```java
+Direction direction = leverBlockState.getValue(LeverBlock.FACING);
```

Comment:
Comment to reiterate that `FACING` references the `Direction` property on the `BlockState`